src/pages/api/locale/index.ts

```diff
@@ -0,0 +1,142 @@
+import { AVAILABLE_LANGUAGE } from "@src/utils/get-locale";
+
+const getHandler: Handler = (request, response): Promise<void> => {
+  const webErrors = {
+    databaseUnreachable: ERRORS_DATA.DATABASE_UNREACHABLE,
+  };
+
+  return getTracer().span("get-locale handler", async (span) => {
+    const userCookie = await getServerSideCookies<UserCookie>(request, {
+      cookieName: "user-cookie",
+      isCookieSealed: true,
+      cookieSalt: configVariables.cookieSalt,
+    });
+
+    if (!userCookie) {
+      span.setDisclosedAttribute("user cookie found", false);
+      throw new NoUserCookieFoundError();
+    }
+    span.setDisclosedAttribute("user cookie found", true);
+
+    const currentDBUser = await getDBUserFromSub(userCookie.sub).catch(
+      (error) => {
+        span.setDisclosedAttribute("user locale retrieved", false);
+        throw webErrorFactory({
+          ...webErrors.databaseUnreachable,
+          parentError: error,
+        });
+      },
+    );
+
+    if (!currentDBUser) {
+      span.setDisclosedAttribute("user locale value", null);
+      response.statusCode = HttpStatus.TEMPORARY_REDIRECT;
+      response.setHeader("location", "/");
+      response.end();
+      return;
+    }
+
+    span.setDisclosedAttribute("user locale retrieved", true);
+    span.setDisclosedAttribute("user locale value", currentDBUser.locale);
+    response.statusCode = HttpStatus.OK;
+    response.json({ locale: currentDBUser.locale });
+    return;
+  });
+};
+
+const postHandler: Handler = (request, response): Promise<void> => {
+  const webErrors = {
+    badRequest: ERRORS_DATA.BAD_REQUEST,
+    databaseUnreachable: ERRORS_DATA.DATABASE_UNREACHABLE,
+  };
+
+  return getTracer().span("set-locale handler", async (span) => {
+    const userCookie = await getServerSideCookies<UserCookie>(request, {
+      cookieName: "user-cookie",
+      isCookieSealed: true,
+      cookieSalt: configVariables.cookieSalt,
+    });
+
+    if (!userCookie) {
+      span.setDisclosedAttribute("user cookie found", false);
+      throw new NoUserCookieFoundError();
+    }
+    span.setDisclosedAttribute("user cookie found", true);
+
+    const currentDBUser = await getDBUserFromSub(userCookie.sub);
+    const { locale } = request.body;
+
+    if (!locale) {
+      throw webErrorFactory(webErrors.badRequest);
+    }
+
+    await getAndPutUser({
+      ...(currentDBUser as DynamoUser),
+      locale: locale as string,
+    })
+      .then(() =>
+        setAlertMessagesCookie(response, [
+          generateAlertMessage(
+            `Your language has been set to ${AVAILABLE_LANGUAGE[locale]}`,
+          ),
+        ]),
+      )
+      .catch((error) => {
+        span.setDisclosedAttribute("user locale set", false);
+        throw webErrorFactory({
+          ...webErrors.databaseUnreachable,
+          parentError: error,
+        });
+      });
+
+    span.setDisclosedAttribute("user locale set", true);
+    response.statusCode = HttpStatus.OK;
```

Comment:
If we use a `POST` method, it should be
```suggestion
    response.statusCode = HttpStatus.CREATED;
```